helm/templates/certificate_webhook-conf.yaml

```diff
@@ -13,19 +13,98 @@ metadata:
     {{- include "helm.labels" . | nindent 4 }}
 type: Opaque
 data:
-  tls.crt: {{ $certs.Cert | b64enc }}
-  tls.key: {{ $certs.Key | b64enc }}
+  tls.crt: {{ default ($certs.Cert | b64enc) (include "getInstalledTLSCert" .) }}
+  tls.key: {{ default ($certs.Key | b64enc) (include "getInstalledTLSKey" .) }}
 ---
-apiVersion: v1
-kind: ConfigMap
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
 metadata:
-  name: {{ .Chart.Name }}-script
-  namespace: {{ .Release.Namespace }}
+  name: {{ .Chart.Name }}-webhook
+  labels:
+    {{- include "helm.labels" . | nindent 4 }}
+webhooks:
+  - name: {{ .Chart.Name }}-svc.{{ .Release.Namespace }}.svc
+    {{- if .Values.deployment.failurePolicy }}
+    failurePolicy: {{ .Values.deployment.failurePolicy }}
+    {{- else }}
+    failurePolicy: Fail
+    {{- end }}
+    clientConfig:
+      service:
+        name: {{ .Chart.Name }}-svc
+        namespace: {{ .Release.Namespace }}
+        path: /mutate
+      caBundle: {{ default ($certs.Cert | b64enc) (include "getInstalledTLSCert" .) }}
+    rules: []
+    sideEffects: None
+    admissionReviewVersions: ["v1beta1"]
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: {{ .Chart.Name }}-webhook
   labels:
     {{- include "helm.labels" . | nindent 4 }}
   annotations:
-    helm.sh/hook: post-install, post-upgrade
-    helm.sh/hook-delete-policy: hook-succeeded, hook-failed
-data:
-  webhook.yaml: |
-    {{- include "webhooks.yaml.tpl" (dict "context" . "cert" $certs.Cert) | indent 4}}
+    "helm.sh/hook": pre-upgrade,pre-rollback
+webhooks:
+  - name: {{ .Chart.Name }}-svc.{{ .Release.Namespace }}.svc
+    failurePolicy: Fail
```

Comment:
is it common to leave this at fail for upgrade? seems risky though no rule would make any call necessary. One could at this point hook in connaisseur with a `RollingUpdate` Policy and keep it functional throughout the upgrade :shrug: 
if we do with turning it off, we'd be interrupting operation